examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3591,25 +3591,65 @@ class GLTFParser {
 	/**
 	 * Specification: https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#skins
 	 * @param {number} skinIndex
-	 * @return {Promise<Object>}
+	 * @return {Promise<Skeleton>}
 	 */
 	loadSkin( skinIndex ) {
 
 		const skinDef = this.json.skins[ skinIndex ];
 
-		const skinEntry = { joints: skinDef.joints };
+		const pending = [];
+
+		for ( let i = 0, il = skinDef.joints.length; i < il; i ++ ) {
+
+			pending.push( this.getDependency( 'node', skinDef.joints[ i ] ) );
+
+		}
+
+		if ( skinDef.inverseBindMatrices !== undefined ) {
+
+			pending.push( this.getDependency( 'accessor', skinDef.inverseBindMatrices ) );
 
-		if ( skinDef.inverseBindMatrices === undefined ) {
+		} else {
 
-			return Promise.resolve( skinEntry );
+			pending.push( null );
 
 		}
 
-		return this.getDependency( 'accessor', skinDef.inverseBindMatrices ).then( function ( accessor ) {
+		return Promise.all( pending ).then( function ( results ) {
+
+			const inverseBindMatrices = results.pop();
+			const jointNodes = results;
+
+			const bones = [];
```

Comment:
Any reason we couldn't initialize `const bones = results;` here? I see we had an exception for this before, but I've never encountered that warning, and it'd be a deeply invalid armature in that situation...

Reply:
Yeah, I don't think I have seen the warning and agree with that it's invalid armature.

But just in case I hope we can make a separate PR for it if needed. I don't really want to add changes that can break anything in a refactoring PR.